pkg/config/config.go

```diff
@@ -198,53 +197,66 @@ type UpdateSetting struct {
 	IncludeDiff bool     `yaml:"includeDiff"`
 }
 
-// Namespaces contains namespaces to include and ignore
-// Include contains a list of namespaces to be watched,
-//  - "all" to watch all the namespaces
-// Ignore contains a list of namespaces to be ignored when all namespaces are included
-// It is an optional (omitempty) field which is tandem with Include [all]
-// It can also contain a * that would expand to zero or more arbitrary characters
-// example : include [all], ignore [x,y,secret-ns-*]
+// Namespaces provides an option to include and exclude given Namespaces.
 type Namespaces struct {
+	// Include contains a list of allowed Namespaces.
+	// It can also contain a regex expressions:
+	//  - ".*" - to specify all Namespaces.
 	Include []string `yaml:"include"`
-	Ignore  []string `yaml:"ignore,omitempty"`
+
+	// Exclude contains a list of Namespaces to be ignored even if allowed by Include.
+	// It can also contain a regex expressions:
+	//  - "test-.*" - to specif all Namespaces with `test-` prefix.
+	Exclude []string `yaml:"exclude,omitempty"`
 }
 
 // IsAllowed checks if a given Namespace is allowed based on the config.
-// Copied from https://github.com/kubeshop/botkube/blob/b6b7d449278617d40f05d0792b419a7692ad980f/pkg/filterengine/filters/namespace_checker.go#L54-L76
-// TODO(https://github.com/kubeshop/botkube/issues/596): adjust contract.
 func (n *Namespaces) IsAllowed(givenNs string) bool {
```

Comment:
I'm not sure if that's a good place (I don't think so), but will you also apply the last AC besides documentation?

> add validation to print warning when include matches all namespaces but on the list there are more items. Because in this case, we will match all Namespaces anyway.

You can definitely push a commit here as I don't expect this change to be that big 🙂

Reply:
OK, I thought so at the beginning but then, it turns out that the impl is not so straight forward, but here it is: https://github.com/kubeshop/botkube/pull/686/commits/93b5fdcf409e004f8e7687f9d038061f40ab632a